src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1821,6 +1829,48 @@ public static boolean fromTheSameCaseExpression(@NotNull PsiElement origin, @Not
     return false;
   }
 
+  public static class ErlangFunctionCallNamedParameter<T extends PsiElement> extends PatternCondition<T> {
+    private final List<String> NAMES;
+
+    public ErlangFunctionCallNamedParameter(@NotNull String... names) {
+      super("functionCallNamedParameter");
+      NAMES = ContainerUtil.newArrayList(names);
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangQAtom argument = element instanceof ErlangQAtom ? (ErlangQAtom) element : null;
+      if (argument == null) return false;
+
+      ErlangFunctionCallExpression call = PsiTreeUtil.getParentOfType(argument, ErlangFunctionCallExpression.class);
```

Comment:
I mean it's recursive in a way it works - it loops through all of the parents, whereas you only need to check like a parent, or, may be a grandparent.

Take a look at how it's done in another pattern which was used before your changes.

Just so you comprehend the problem here: ```foo({1, {2, .....{N, atom}}}}}})``` will make you traverse all the way up through N levels of nested tuples to get to the function call expression.